File: leetcode/2578.最小和分割.go
func splitNum(num int) int {
    stnum := []byte(strconv.Itoa(num))
    sort.Slice(stnum, func(i, j int) bool {
        return stnum[i] < stnum[j]
    })
    num1, num2 := 0, 0
    for i := 0; i < len(stnum); i++ {
        if i % 2 == 0 {
            num1 = num1 * 10 + int(stnum[i] - '0')
        } else {
            num2 = num2 * 10 + int(stnum[i] - '0')
        }
    }
    return num1 + num2
}

File: leetcode/2578.最小和分割.py
class Solution:
    def splitNum(self, num: int) -> int:
        num = [c for c in str(num)]
        num.sort()
        odd = []
        even = []
        for i, c in enumerate(num):
            if i % 2:
                odd.append(c)
            else:
                even.append(c)
        return int("".join(odd)) + int("".join(even))